src/fuselage/CCPACSFuselageProfile.cpp

```diff
@@ -175,71 +175,85 @@ TopoDS_Wire CCPACSFuselageProfile::GetWire(bool forceClosed) const
-     }
-
-    // Create wires
-    TopoDS_Edge edge = BRepBuilderAPI_MakeEdge(spline).Edge();
-    BRepBuilderAPI_MakeWire builder1(edge);
-    TopoDS_Wire tempWireOriginal = builder1.Wire();
-
-    BRepBuilderAPI_MakeWire builder2(edge);
-    if (!spline->IsClosed()) {
-        builder2.Add(BRepBuilderAPI_MakeEdge(spline->EndPoint(), spline->StartPoint()));
-    }
-    TopoDS_Wire tempWireClosed = builder2.Wire();
-    if (tempWireClosed.IsNull() == Standard_True || tempWireOriginal.IsNull() == Standard_True) {
-        throw CTiglError("TopoDS_Wire is null in CCPACSFuselageProfile::BuildWire", TIGL_ERROR);
-    }
+        // Here, the B-spline is reparameterized based on the CPACS profile after setting it up at first for accuracy reasons
+        // Also, a tolerance is passed used for knot insertion and removal during the reparameterization algorithm
+        CTiglInterpolatePointsWithKinks interp(occPoints, kinks, params, 0.5, 3, CTiglInterpolatePointsWithKinks::Algo::InterpolateFirstThenReparametrize, 1e-8);
+        auto spline = interp.Curve();
+
+        if (mirrorSymmetry) {
+            double umin = spline->FirstParameter();
+            double umax = spline->LastParameter();
+            spline      = CTiglBSplineAlgorithms::trimCurve(spline, umin, 0.5 * (umin + umax));
+            CTiglBSplineAlgorithms::reparametrizeBSpline(*spline, umin, umax);
+        }
+
+        // Reparamaterization based on a ParamMap defined in the CPACS file within CTiglInterpolatePointsWithKinks does not get along with reparametrizeBSplineNiceKnots.
+        // The geometry is changed in a way that is not acceptable anymore. However out of performance reasons, the feature should not be rejected in the most cases.
+        // Due to those conflicts, the function is only called, when there are no parameters defined in the CPACS file:
+        if (params.empty()) {
+            // we reparametrize the spline to get better performing lofts.
+            // there might be a small accuracy loss though.
+            spline = CTiglBSplineAlgorithms::reparametrizeBSplineNiceKnots(spline).curve;
+         }
+
+        // Create wires
+        TopoDS_Edge edge = BRepBuilderAPI_MakeEdge(spline).Edge();
+        BRepBuilderAPI_MakeWire builder1(edge);
+        TopoDS_Wire tempWireOriginal = builder1.Wire();
+
+        BRepBuilderAPI_MakeWire builder2(edge);
+        if (!spline->IsClosed()) {
+            builder2.Add(BRepBuilderAPI_MakeEdge(spline->EndPoint(), spline->StartPoint()));
+        }
+        TopoDS_Wire tempWireClosed = builder2.Wire();
+        if (tempWireClosed.IsNull() == Standard_True || tempWireOriginal.IsNull() == Standard_True) {
+            throw CTiglError("TopoDS_Wire is null in CCPACSFuselageProfile::BuildWire", TIGL_ERROR);
+        }
 
-    cache.closed   = tempWireClosed;
-    cache.original = tempWireOriginal;
+        cache.closed   = tempWireClosed;
+        cache.original = tempWireOriginal;
 }
 
```

Comment:
Accidentally deleted Implementation of BuildWiresRectangle

```suggestion

//Builds the fuselage profile wire from heightToWidthRatio and cornerRadius with a tolerance of 1e-3 for the wire
void CCPACSFuselageProfile::BuildWiresRectangle(WireCache& cache) const
{
    if(!m_standardProfile_choice3->GetRectangle_choice1()){
        throw CTiglError("CCPACSFuselageProfile::BuildWire", TIGL_ERROR);
    }
    //Get Paramenters
    auto& rectangle_profile = *m_standardProfile_choice3->GetRectangle_choice1();
    double heightToWidthRatio = rectangle_profile.GetHeightToWidthRatio().GetValue();
    double radius = (rectangle_profile.GetCornerRadius())? *rectangle_profile.GetCornerRadius() : 0. ;
    //Build wire
    TopoDS_Wire wire = BuildWireRectangle(heightToWidthRatio,radius, 1e-3);
    cache.closed = wire;
    cache.original = wire;
}

```